cmd/export/main.go

```diff
@@ -17,37 +17,102 @@ package main
 
 import (
 	"context"
-	"log"
+	"fmt"
 	"net/http"
+	"time"
 
 	"go.opencensus.io/plugin/ochttp"
 
 	"github.com/google/exposure-notifications-server/internal/export"
 	"github.com/google/exposure-notifications-server/internal/logging"
 	_ "github.com/google/exposure-notifications-server/internal/observability"
+	"github.com/google/exposure-notifications-server/internal/serverenv"
 	"github.com/google/exposure-notifications-server/internal/setup"
 )
 
 func main() {
 	ctx := context.Background()
 	logger := logging.FromContext(ctx)
 
+	if err := realMain(); err != nil {
+		logger.Fatal(err)
+	}
+}
+
+func realMain() error {
+	ctx := context.Background()
+
+	config, env, closer, err := doSetup()
+	if err != nil {
+		return fmt.Errorf("setup: %w", err)
+	}
+	defer closer()
```

Comment:
Having this defer in `main` will _never_ run because `logger.Fatal` calls `os.Exit` and that doesn't run defers.

Reply:
known issue.
Also cloud run stops container instances with an un-trappable signal.

This is there in case someone wanted to let this have a more graceful shutdown.